File: Python/dictionary/Max Number of K-Sum Pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        mp={}
        cnt=0
        for i in range(len(nums)):
            comp = k-nums[i]
            if comp in mp and mp[comp]>0:
                mp[comp]-=1
                cnt+=1
            else:
                if nums[i] not in mp:
                    mp[nums[i]]=0
                mp[nums[i]]+=1
        return cnt
        

File: CPP/map/Max Number of K-Sum Pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        unordered_map<int,int>mp;
        int cnt =0;
        for(auto i:nums){
            int find = k-i;
            if(mp[find]>0){
                mp[find]--;
                cnt++;
            }
            else
                mp[i]++;
        }
        return cnt;        
    }
};
